src/convertToObject.js

```diff
@@ -3,16 +3,36 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
+const PROPERTY_SEPARATOR = ';';
+const PROPERTY_VALUE_SEPARATOR = ':';
+
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString
+    .split(PROPERTY_SEPARATOR)
+    .reduce((prev, styleStr) => {
+      const style = styleStr.split(PROPERTY_VALUE_SEPARATOR);
+
+      if (style[0].trim() && style[1].trim()) {
+        const styleName = style[0].trim();
+        const styleValue = style[1].trim();
```

Comment:
```suggestion
      const styleName = style[0].trim();
      const styleValue = style[1].trim();
      
      if (styleName && styleValue) {
```

Reply:
Why you can't simplify it like this?

Reply:
style[1] can be undefined, so there is an error cannot trim() undefined, so I can't do this

Reply:
I've figured it out!